### PR TITLE
EOS-27018: Hare to adopt utils logging framework

### PR DESCRIPTION
**Pull request by SwapnilGaonkar7**
Problem: MessageBUs class is singleton class and component need to
call MessageBus.init method before making use of Message Bus library

Solution: Calling MessageBus.init method in 'start' interface before
starting other Hare related services

Tested by Manual installation of RPMs.

**Deployment is in progress.**

Signed-off-by: Swapnil Gaonkar <swapnil.gaonkar@seagate.com>

### CODE REVIEW

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -383,8 +384,15 @@ def start_hax_and_consul_without_systemd(url: str, utils: Utils):
...
 
 def start(args):
+    logging.info('Starting Hare services')
     url = args.config[0]
+    provider = ConfStoreProvider(url)
     utils = Utils(ConfStoreProvider(url))
+
+    message_server_endpoints = provider.get('cortx>external>kafka>endpoints')
+    MessageBus.init(message_server_endpoints)
+    logging.info('message_server_endpoints = %s', message_server_endpoints)
```
Comment (sachinpunadikar) [inline provisioning/miniprov/hare_mp/main.py:394]: This can be skipped.

**Review by sachinpunadikar — APPROVED** [review]
+1

File: hax/hax/ha/events.py [hunk 2]
```diff
@@ -75,9 +77,15 @@ def __init__(self,
...
             raise RuntimeError('Failed to subscribe to events')
 
-        message_bus = MessageBus()
+        util: ConsulUtil = ConsulUtil()
+        configpath = util.get_configpath()
+        Conf.load('cortx_conf', configpath)
+
+        message_server_endpoints = Conf.get('cortx_conf',
+                                            'cortx>external>kafka>endpoints')
+        MessageBus.init(message_server_endpoints)
```
Comment (mssawant) [inline hax/hax/ha/events.py:86]: Just add some documentation here saying,
```
This creates a MessageBus internal global in-memory object which is indirectly part of the process address space.
Thus `MessageBus.init()` must be invoked within process's address space.
```

Comment (SwapnilGaonkar7) [reply hax/hax/ha/events.py:92]: Done

**Comment by SwapnilGaonkar7** [comment]
@mssawant Addressed review comment please have a look

**Review by suvratjoshi — APPROVED** [review]

File: hax/hax/ha/events.py [hunk 3]
```diff
@@ -75,9 +77,21 @@ def __init__(self,
...
+
+        message_server_endpoints = Conf.get('cortx_conf',
+                                            'cortx>external>kafka>endpoints')
+
+        """
+        This creates a MessageBus internal global in-memory object which is
+        indirectly part of the process address space. Thus
+        `MessageBus.init()` must be invoked within process's address space.
+        """
+        MessageBus.init(message_server_endpoints)
```
Comment (mssawant) [inline hax/hax/ha/events.py:92]: @SwapnilGaonkar7 , I think this should be moved to hax.py::main(), please introduce a new function, ha_message_bus_init() and move the code in it.

**Comment by SwapnilGaonkar7** [comment]
Tested regular(**non redefined**) deployment using http://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/centos-7.9/job/cortx-all-image-custom-ci/1765/
Also tested Message Consumer by sending events using https://raw.githubusercontent.com/Seagate/cortx-hare/main/ha-simulator/prod_consumer.py

**Pull request merged**